network/src/commonMain/kotlin/com/wire/kalium/network/api/conversation/ConversationApiImpl.kt

```diff
@@ -77,16 +69,38 @@ class ConversationApiImpl internal constructor(private val authenticatedNetworkC
     /**
      * returns 200 conversation created or 204 conversation unchanged
      */
-    override suspend fun addParticipant(
-        addParticipantRequest: AddParticipantRequest,
+    override suspend fun addMember(
+        request: AddConversationMembersRequest,
         conversationId: ConversationId
-    ): NetworkResponse<AddParticipantResponse> = try {
+    ): NetworkResponse<ConversationMemberChangeDTO> = try {
         httpClient.post("$PATH_CONVERSATIONS/${conversationId.value}/$PATH_MEMBERS/$PATH_V2") {
-            setBody(addParticipantRequest)
+            setBody(request)
         }.let { response ->
             when (response.status) {
-                HttpStatusCode.OK -> wrapKaliumResponse<AddParticipantResponse.UserAdded> { response }
-                HttpStatusCode.NoContent -> wrapKaliumResponse<AddParticipantResponse.ConversationUnchanged> { response }
+                HttpStatusCode.OK -> wrapKaliumResponse<ConversationMemberChangeDTO.Added> { response }
+                // TODO should we show user some message that there was no change?
+                HttpStatusCode.NoContent -> NetworkResponse.Success(ConversationMemberChangeDTO.Unchanged, response)
+                else -> wrapKaliumResponse { response }
+            }
+        }
+    } catch (e: IOException) {
+        NetworkResponse.Error(KaliumException.GenericError(e))
+    }
+
+    /**
+     * returns 200 Member removed and 204 No change
+     */
+    override suspend fun removeMember(
+        userId: UserId,
+        conversationId: ConversationId
+    ): NetworkResponse<ConversationMemberChangeDTO> = try {
+        httpClient.delete(
+            "$PATH_CONVERSATIONS/${conversationId.domain}/${conversationId.value}/$PATH_MEMBERS/${userId.domain}/${userId.value}"
+        ).let { response ->
+            when (response.status) {
+                HttpStatusCode.OK -> wrapKaliumResponse<ConversationMemberChangeDTO.Removed> { response }
+                // TODO should we show user some message that there was no change?
```

Comment:
I'd say no, this is a race condition when two users perform the same action which resolves it self so the user doesn't need to retry his action.